openedx/features/course_experience/api/v1/tests/test_views.py

```diff
@@ -25,17 +27,22 @@ def setUp(self):  # pylint: disable=arguments-differ
         # Need to supply tracker name for the EventTestMixin. Also, EventTestMixin needs to come
         # first in class inheritance so the setUp call here appropriately works
         super().setUp('openedx.features.course_experience.api.v1.views.tracker')
+        self.course = CourseFactory.create(self_paced=True, start=timezone.now() - datetime.timedelta(days=1000))
 
     def test_reset_deadlines(self):
-        CourseEnrollment.enroll(self.user, self.course.id, CourseMode.VERIFIED)
+        enrollment = CourseEnrollment.enroll(self.user, self.course.id, CourseMode.VERIFIED)
+        enrollment.schedule.start_date = timezone.now() - datetime.timedelta(days=100)
+        enrollment.schedule.save()
         # Test body with incorrect body param (course_key is required)
         response = self.client.post(reverse('course-experience-reset-course-deadlines'), {'course': self.course.id})
         assert response.status_code == 400
+        assert enrollment.schedule == Schedule.objects.get(id=enrollment.schedule.id)
         self.assert_no_events_were_emitted()
 
         # Test correct post body
         response = self.client.post(reverse('course-experience-reset-course-deadlines'), {'course_key': self.course.id})
         assert response.status_code == 200
+        assert enrollment.schedule.start_date < Schedule.objects.get(id=enrollment.schedule.id).start_date
```

Comment:
This test wasn't working correctly before. The course wasn't self-paced, so the schedule [was never modified](https://github.com/openedx/edx-platform/blob/656ec5def9e211ca6b41f5ec3fd5764dd7dc3c3e/openedx/core/djangoapps/schedules/utils.py#L54).